compaction/leveled_compaction_strategy.cc

```diff
@@ -47,8 +47,8 @@ compaction_descriptor leveled_compaction_strategy::get_sstables_for_compaction(t
             continue;
         }
         auto& sst = *std::max_element(sstables.begin(), sstables.end(), [&] (auto& i, auto& j) {
-            auto gc_before1 = i->get_gc_before_for_drop_estimation(compaction_time);
-            auto gc_before2 = j->get_gc_before_for_drop_estimation(compaction_time);
+            auto gc_before1 = i->get_gc_before_for_drop_estimation(compaction_time, get_tombstone_gc_state());
+            auto gc_before2 = j->get_gc_before_for_drop_estimation(compaction_time, get_tombstone_gc_state());
```

Comment:
If you want to avoid the burden of passing tombstone_gc_state() in some places, you can define table_state::tombstone_gc_state(), which can in turn retrieve it from table's compaction manager. Then unit tests can have a table state implemented which retrieves the state that can be stored anywhere.

Reply:
Absolutely, table_state was designed for this.

Reply:
this is simpler and more elegant indeed.